### ISSUE TITLE
Test failure: `example.armeria.grpc.kotlin.HelloServiceTest`

### ISSUE THREAD

**Issue by ikhoon** [line/armeria#4367]
```
io.grpc.StatusException: UNKNOWN
	at app//io.grpc.Status.asException(Status.java:550)
	at app//io.grpc.kotlin.ClientCalls$rpcImpl$1$1$1.onClose(ClientCalls.kt:296)
	at app//com.linecorp.armeria.internal.client.grpc.ArmeriaClientCall.close(ArmeriaClientCall.java:499)
	at app//com.linecorp.armeria.internal.client.grpc.ArmeriaClientCall.transportReportStatus(ArmeriaClientCall.java:443)
	at app//com.linecorp.armeria.internal.common.grpc.GrpcStatus.reportStatus(GrpcStatus.java:338)
	at app//com.linecorp.armeria.internal.common.grpc.GrpcStatus.lambda$reportStatusLater$0(GrpcStatus.java:316)
	at java.base@16.0.2/java.util.concurrent.CompletableFuture.uniHandle(CompletableFuture.java:930)
	at java.base@16.0.2/java.util.concurrent.CompletableFuture$UniHandle.tryFire(CompletableFuture.java:907)
	at java.base@16.0.2/java.util.concurrent.CompletableFuture.postComplete(CompletableFuture.java:506)
	at java.base@16.0.2/java.util.concurrent.CompletableFuture.complete(CompletableFuture.java:2143)
	at app//com.linecorp.armeria.common.stream.AbstractStreamMessage$CloseEvent.notifySubscriber(AbstractStreamMessage.java:259)
	at app//com.linecorp.armeria.common.stream.DefaultStreamMessage.notifySubscriberOfCloseEvent0(DefaultStreamMessage.java:300)
	at app//com.linecorp.armeria.common.stream.DefaultStreamMessage.notifySubscriberOfCloseEvent(DefaultStreamMessage.java:292)
	at app//com.linecorp.armeria.common.stream.DefaultStreamMessage.handleCloseEvent(DefaultStreamMessage.java:429)
	at app//com.linecorp.armeria.common.stream.DefaultStreamMessage.notifySubscriber0(DefaultStreamMessage.java:372)
	at app//com.linecorp.armeria.common.stream.DefaultStreamMessage.notifySubscriber(DefaultStreamMessage.java:328)
	at app//com.linecorp.armeria.common.stream.DefaultStreamMessage.addObjectOrEvent(DefaultStreamMessage.java:314)
	at app//com.linecorp.armeria.common.stream.DefaultStreamMessage.close(DefaultStreamMessage.java:436)
	at app//com.linecorp.armeria.internal.common.stream.DecodedStreamMessage$DecodingSubscriber.onComplete(DecodedStreamMessage.java:288)
	at app//com.linecorp.armeria.common.stream.AbstractStreamMessage$CloseEvent.notifySubscriber(AbstractStreamMessage.java:258)
	at app//com.linecorp.armeria.common.stream.DefaultStreamMessage.notifySubscriberOfCloseEvent0(DefaultStreamMessage.java:300)
	at app//com.linecorp.armeria.common.stream.DefaultStreamMessage.notifySubscriberOfCloseEvent(DefaultStreamMessage.java:292)
	at app//com.linecorp.armeria.common.stream.DefaultStreamMessage.handleCloseEvent(DefaultStreamMessage.java:429)
	at app//com.linecorp.armeria.common.stream.DefaultStreamMessage.notifySubscriber0(DefaultStreamMessage.java:372)
	at app//com.linecorp.armeria.common.stream.DefaultStreamMessage.notifySubscriber(DefaultStreamMessage.java:328)
	at app//com.linecorp.armeria.common.stream.DefaultStreamMessage.addObjectOrEvent(DefaultStreamMessage.java:314)
	at app//com.linecorp.armeria.common.stream.DefaultStreamMessage.close(DefaultStreamMessage.java:436)
	at app//com.linecorp.armeria.client.HttpResponseDecoder$HttpResponseWrapper.closeAction(HttpResponseDecoder.java:335)
	at app//com.linecorp.armeria.client.HttpResponseDecoder$HttpResponseWrapper.cancelTimeoutOrLog(HttpResponseDecoder.java:370)
	at app//com.linecorp.armeria.client.HttpResponseDecoder$HttpResponseWrapper.close(HttpResponseDecoder.java:318)
	at app//com.linecorp.armeria.client.HttpResponseDecoder$HttpResponseWrapper.close(HttpResponseDecoder.java:308)
	at app//com.linecorp.armeria.client.Http2ResponseDecoder.onHeadersRead(Http2ResponseDecoder.java:216)
	at app//com.linecorp.armeria.client.Http2ResponseDecoder.onHeadersRead(Http2ResponseDecoder.java:225)
	at app//io.netty.handler.codec.http2.Http2FrameListenerDecorator.onHeadersRead(Http2FrameListenerDecorator.java:48)
	at app//io.netty.handler.codec.http2.Http2EmptyDataFrameListener.onHeadersRead(Http2EmptyDataFrameListener.java:63)
	at app//io.netty.handler.codec.http2.DefaultHttp2ConnectionDecoder$FrameReadListener.onHeadersRead(DefaultHttp2ConnectionDecoder.java:409)
	at app//io.netty.handler.codec.http2.DefaultHttp2ConnectionDecoder$FrameReadListener.onHeadersRead(DefaultHttp2ConnectionDecoder.java:337)
	at app//io.netty.handler.codec.http2.Http2InboundFrameLogger$1.onHeadersRead(Http2InboundFrameLogger.java:56)
	at app//io.netty.handler.codec.http2.DefaultHttp2FrameReader$2.processFragment(DefaultHttp2FrameReader.java:476)
	at app//io.netty.handler.codec.http2.DefaultHttp2FrameReader.readHeadersFrame(DefaultHttp2FrameReader.java:484)
	at app//io.netty.handler.codec.http2.DefaultHttp2FrameReader.processPayloadState(DefaultHttp2FrameReader.java:253)
	at app//io.netty.handler.codec.http2.DefaultHttp2FrameReader.readFrame(DefaultHttp2FrameReader.java:159)
	at app//io.netty.handler.codec.http2.Http2InboundFrameLogger.readFrame(Http2InboundFrameLogger.java:41)
	at app//io.netty.handler.codec.http2.DefaultHttp2ConnectionDecoder.decodeFrame(DefaultHttp2ConnectionDecoder.java:173)
	at app//io.netty.handler.codec.http2.DecoratingHttp2ConnectionDecoder.decodeFrame(DecoratingHttp2ConnectionDecoder.java:63)
	at app//io.netty.handler.codec.http2.Http2ConnectionHandler$FrameDecoder.decode(Http2ConnectionHandler.java:393)
	at app//io.netty.handler.codec.http2.Http2ConnectionHandler.decode(Http2ConnectionHandler.java:453)
	at app//io.netty.handler.codec.ByteToMessageDecoder.decodeRemovalReentryProtection(ByteToMessageDecoder.java:510)
	at app//io.netty.handler.codec.ByteToMessageDecoder.callDecode(ByteToMessageDecoder.java:449)
	at app//io.netty.handler.codec.ByteToMessageDecoder.channelRead(ByteToMessageDecoder.java:279)
	at app//io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:379)
	at app//io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:365)
	at app//io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:357)
	at app//io.netty.handler.logging.LoggingHan
```

The cause of the error above seems suspicious.
```
Caused by: com.linecorp.armeria.common.grpc.StatusCauseException: java.lang.IllegalStateException: subscribed by other subscriber already
	at com.linecorp.armeria.internal.common.stream.FixedStreamMessage.collect(FixedStreamMessage.java:189)
	at com.linecorp.armeria.server.grpc.UnaryServerCall.startDeframing(UnaryServerCall.java:118)
	at com.linecorp.armeria.server.grpc.FramedGrpcService.startCall(FramedGrpcService.java:316)
	at com.linecorp.armeria.server.grpc.FramedGrpcService.lambda$startCall$3(FramedGrpcService.java:284)
	at com.google.common.util.concurrent.SequentialExecutor$1.run(SequentialExecutor.java:123)
	at com.google.common.util.concurrent.SequentialExecutor$QueueWorker.workOnQueue(SequentialExecutor.java:235)
	at com.google.common.util.concurrent.SequentialExecutor$QueueWorker.run(SequentialExecutor.java:180)
	at com.linecorp.armeria.common.RequestContext.lambda$makeContextAware$3(RequestContext.java:555)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515)
	at java.util.concurrent.FutureTask.run(FutureTask.java:264)
```
https://github.com/line/armeria/actions/runs/2738340617#summary-7516955221

**Comment by ikhoon** [comment]
It may be possible that a `FixedStreamMessage` is aborted before being collected by `UnaryServerCall`.

**Comment by jmadureira** [comment]
@ikhoon don't know if this information helps but I'm getting the same error randomly when running tests on my project which is Java based and only if I enable `useBlockingTaskExecutor(true)` on my gRPC service.

**Comment by ikhoon** [comment]
It is super useful. Let me try to reproduce it.

**Comment by ikhoon** [comment]
I couldn't figure out the cause of the regression but it seems related to #4331

**Comment by ikhoon** [comment]
It is difficult to reproduce the failure locally. @jmadureira Do you have some logs produced by `LoggingService` for the error? I'd like to get some request/response duration metrics.

**Comment by jmadureira** [comment]
@ikhoon below is an example from one of our test runs:
```
2022-07-28T08:01:53,233Z [armeria-common-worker-nio-2-3] WARN  c.l.a.server.logging.LoggingService [traceId=] - [sreqId=1ca9b4ae, chanId=1e8d59c6, raddr=127.0.0.1:50359, laddr=127.0.0.1:50357][h2c://qf20y2r2fk/c.s.s.grpc.v1.PermissionService/WriteRelationships#POST] Request: {startTime=2022-07-28T08:01:53.228Z(1658995313228958), length=229B, duration=4297µs(4297333ns), cause=java.lang.IllegalStateException: subscribed by other subscriber already, scheme=gproto+h2c, name=WriteRelationships, headers=[:method=POST, :path=/c.s.s.grpc.v1.PermissionService/WriteRelationships, :scheme=http, :authority=localhost:50357, content-type=application/grpc+proto, te=trailers, grpc-accept-encoding=gzip, grpc-timeout=15000000u, authorization=Bearer ey...eu4bL, user-agent=armeria/1.17.1, content-length=229], content=DefaultRpcRequest{serviceType=GrpcLogUtil, serviceName=c.s.s.grpc.v1.PermissionService, method=WriteRelationships, params=[]}}
2022-07-28T08:01:53,233Z [armeria-common-worker-nio-2-3] WARN  c.l.a.server.logging.LoggingService [traceId=] - [sreqId=1ca9b4ae, chanId=1e8d59c6, raddr=127.0.0.1:50359, laddr=127.0.0.1:50357][h2c://qf20y2r2fk/c.s.s.grpc.v1.PermissionService/WriteRelationships#POST] Response: {startTime=2022-07-28T08:01:53.233Z(1658995313233405), length=0B, duration=215µs(215250ns), totalDuration=4662µs(4662666ns), cause=io.grpc.StatusException: UNKNOWN, headers=[EOS, :status=200, content-type=application/grpc+proto, grpc-encoding=identity, grpc-accept-encoding=gzip, grpc-status=2, content-length=0], content=CompletableRpcResponse{cause=io.grpc.StatusException: UNKNOWN}}
io.grpc.StatusException: UNKNOWN
        at io.grpc.Status.asException(Status.java:543)
        at com.linecorp.armeria.internal.common.grpc.GrpcLogUtil.rpcResponse(GrpcLogUtil.java:61)
        at com.linecorp.armeria.server.grpc.UnaryServerCall.doClose(UnaryServerCall.java:206)
        at com.linecorp.armeria.server.grpc.AbstractServerCall.doClose(AbstractServerCall.java:254)
        at com.linecorp.armeria.server.grpc.AbstractServerCall.close(AbstractServerCall.java:219)
        at com.linecorp.armeria.server.grpc.AbstractServerCall.close(AbstractServerCall.java:209)
        at com.linecorp.armeria.server.grpc.AbstractServerCall.onError(AbstractServerCall.java:419)
        at com.linecorp.armeria.server.grpc.UnaryServerCall.lambda$startDeframing$0(UnaryServerCall.java:120)
        at java.base/java.util.concurrent.CompletableFuture.uniHandle(CompletableFuture.java:934)
        at java.base/java.util.concurrent.CompletableFuture.uniHandleStage(CompletableFuture.java:950)
        at java.base/java.util.concurrent.CompletableFuture.handle(CompletableFuture.java:2340)
        at com.linecorp.armeria.server.grpc.UnaryServerCall.startDeframing(UnaryServerCall.java:118)
        at com.linecorp.armeria.server.grpc.UnaryServerCall.request(UnaryServerCall.java:109)
        at com.linecorp.armeria.common.RequestContext.lambda$makeContextAware$3(RequestContext.java:555)
        at io.netty.util.concurrent.AbstractEventExecutor.runTask(AbstractEventExecutor.java:174)
        at io.netty.util.concurrent.AbstractEventExecutor.safeExecute(AbstractEventExecutor.java:167)
        at io.netty.util.concurrent.SingleThreadEventExecutor.runAllTasks(SingleThreadEventExecutor.java:470)
        at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:503)
        at io.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:997)
        at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74)
        at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
        at java.base/java.lang.Thread.run(Thread.java:833)
Caused by: java.lang.IllegalStateException: subscribed by other subscriber already
        at com.linecorp.armeria.internal.common.stream.FixedStreamMessage.collect(FixedStreamMessage.java:189)
        ... 11 common frames omitted
```

I redacted some things for clarity purposes.

I'll see during the day if I can recreate this using a `@RepeatedTest` or something similar.

**Comment by jmadureira** [comment]
@ikhoon actually there's an easier way to recreate this (or expose another issue perhaps).
Enable `useBlockingTaskExecutor(true)` on
https://github.com/line/armeria/blob/7395a2e76459798d0d7c4b6185923d37cb277a97/examples/grpc/src/main/java/example/armeria/grpc/Main.java#L53
... and then use `@org.junit.jupiter.api.RepeatedTest(10000)` instead of `@Test` on
https://github.com/line/armeria/blob/7395a2e76459798d0d7c4b6185923d37cb277a97/examples/grpc/src/test/java/example/armeria/grpc/HelloServiceTest.java#L47

This will run in about 5 seconds and raise ~10 errors or so.

**Comment by ikhoon** [comment]
@jmadureira Your tips were really helpful to find the bug which will be fixed by #4374. Thanks!
